src/Illuminate/Collections/Collection.php

```diff
@@ -10,21 +10,24 @@
 use Illuminate\Support\Traits\Macroable;
 use stdClass;
 
+/**
+ * @template T
```

Comment:
There should be two templates here. `TKey` and `TValue` maybe. So `offset*` methods can also use the key type.

Also in PHPStan (not sure about Psalm) `ArrayAccess` is also generic. So there needs to be `@implements ArrayAccess<TKey, TValue>` here.

Reply:
Yes this crossed my mind but I wanted to keep the scope small.

However, I think it's way more usable if I add this. Great suggestion!

Reply:
@canvural I have made the changes :D